internal/controller/conditions.go

```diff
@@ -170,3 +171,28 @@ func setKubeconfigReadyConditionTrue(cc *v1alpha1.ClusterConnect, message ...str
 		Message: conditionMessage,
 	})
 }
+
+func setConnectionProbeConditionTrue(cc *v1alpha1.ClusterConnect, message ...string) {
+	conditionMessage := ""
+	if len(message) > 0 {
+		conditionMessage = message[0]
+	}
+	v1beta2conditions.Set(cc, metav1.Condition{
+		Type:    v1alpha1.ConnectionProbeCondition,
+		Status:  metav1.ConditionTrue,
+		Reason:  v1alpha1.ConnectionProbeSucceededReason,
+		Message: conditionMessage,
+	})
+}
+func setConnectionProbeConditionFalse(cc *v1alpha1.ClusterConnect, message ...string) {
+	conditionMessage := ""
+	if len(message) > 0 {
+		conditionMessage = message[0]
+	}
+	v1beta2conditions.Set(cc, metav1.Condition{
+		Type:    v1alpha1.ConnectionProbeCondition,
+		Status:  metav1.ConditionFalse,
+		Reason:  v1alpha1.ConnectionProbeFailedReason,
+		Message: conditionMessage,
+	})
+}
```

Comment:
There is lot of commonality here. You may think of making a single function.

Reply:
I meant something like this.

```shell
func setConnectionProbeCondition(cc *v1alpha1.ClusterConnect, status ConditionStatus, reason string, message ...string) {
	conditionMessage := ""
	if len(message) > 0 {
		conditionMessage = message[0]
	}
	v1beta2conditions.Set(cc, metav1.Condition{
		Type:    v1alpha1.ConnectionProbeCondition,
		Status:  status,
		Reason:  reason,
		Message: conditionMessage,
	})
}
```

This single function can now be used instead of separate `setConnectionProbeConditionTrue` and `setConnectionProbeConditionTrue`. Does that make sense?

Reply:
I haven't investigated rest of CCG Controller code, but perhaps `Type` field could be parameterized as well, and this function could be even generalized for setting all types of type/status and not just ConnectionProbeCondition.

Reply:
agree. Let me refactor this code in separate PR